pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -34,29 +41,50 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
 		status = "error"
 	}
 
-	elapsed := time.Since(start) / time.Millisecond
-	pluginRequestDuration.WithLabelValues(pluginID, endpoint).Observe(float64(elapsed))
-	pluginRequestCounter.WithLabelValues(pluginID, endpoint, status).Inc()
+	elapsed := time.Since(start)
+	pluginRequestDuration.WithLabelValues(pluginCtx.PluginID, endpoint).Observe(float64(elapsed / time.Millisecond))
+	pluginRequestCounter.WithLabelValues(pluginCtx.PluginID, endpoint, status).Inc()
+
+	if cfg.LogPluginRequests {
+		logParams := []interface{}{
+			"status", status,
+			"duration", elapsed,
+			"pluginId", pluginCtx.PluginID,
+			"endpoint", endpoint,
+		}
+
+		traceID := tracing.TraceIDFromContext(ctx, false)
```

Comment:
Nit. Could use the contextual logger to automatically get this set.

Reply:
Ah good point. I think we added that after I started this PR :)